File: send_files.py
#!/usr/bin/env python
import os
import pika
import glob

# Параметры подключения к RabbitMQ
rmq_parameters = pika.URLParameters('amqp://artem:321@10.28.0.229:5672/')
connection = pika.BlockingConnection(rmq_parameters)
channel = connection.channel()

# Объявление очереди
channel.queue_declare(queue='scmo_in', durable=True)

# Директория с файлами
output_dir = 'output'

# Функция для отправки файла в RabbitMQ
def send_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as file:
        file_content = file.read()

    file_extension = os.path.splitext(file_path)[1]
    if file_extension == '.xml':
        content_type = 'application/xml'
    elif file_extension == '.json':
        content_type = 'application/json'
    else:
        return

    channel.basic_publish(
        exchange='input',
        routing_key='',
        body=file_content,
        properties=pika.BasicProperties(
            type='Item',
            app_id='erp_in',
            content_type=content_type
        )
    )

    print(f" [x] Sent file {file_path}")

# Получение списка файлов в директории
files = glob.glob(os.path.join(output_dir, '*'))

# Отправка каждого файла
for file_path in files:
    send_file(file_path)

# Закрытие соединения
connection.close()

File: src/connector/sendFiles.js
// src/connector/sendFiles.js

const fs = require('fs').promises;
const path = require('path');
const amqp = require('amqplib');
const { logInfo, logError } = require('../utils/logger');
const { rabbitMQ } = require('../../config');

const outputDir = path.join(__dirname, '../../output'); // путь к директории output

const sendFileToRabbitMQ = async (filePath) => {
    try {
        const conn = await amqp.connect(rabbitMQ.url);
        const ch = await conn.createChannel();
        const queue = 'scmo_in'; // Объявление очереди
        const fileName = path.basename(filePath);
        const fileContent = await fs.readFile(filePath, 'utf8');

        const fileExtension = path.extname(filePath);
        let contentType;
        if (fileExtension === '.xml') {
            contentType = 'application/xml';
        } else if (fileExtension === '.json') {
            contentType = 'application/json';
        } else {
            return; // Пропускаем файлы с неизвестным расширением
        }

        await ch.assertQueue(queue, { durable: true });

        ch.sendToQueue(queue, Buffer.from(fileContent), {
            persistent: true,
            contentType: contentType,
            type: 'Item',
            appId: 'erp_in'
        });

        logInfo(`File ${fileName} sent to RabbitMQ queue ${queue}`);

        setTimeout(() => {
            conn.close();
        }, 500);
    } catch (error) {
        logError(`Error sending file ${filePath}: ${error.message}`);
    }
};

const sendFiles = async () => {
    try {
        const files = await fs.readdir(outputDir);

        for (const file of files) {
            const filePath = path.join(outputDir, file);
            await sendFileToRabbitMQ(filePath);
        }
    } catch (error) {
        logError(`Error reading output directory: ${error.message}`);
    }
};

sendFiles();
